recipe-server/normandy/settings.py

```diff
@@ -153,6 +153,20 @@ def CSP_DEFAULT_SRC(self):
         'opt-out-study': os.path.join(BASE_DIR, 'client/actions/opt-out-study'),
     }
 
+    # Mapping of named actions that don't exist in the repo but exists
+    # in the installed mozilla-normandy-action-argument-schemas node package.
+    REMOTE_ACTIONS = {
+        # Name we use: Name inside mozilla-normandy-action-argument-schemas
+        'console-log2': 'consoleLog',
+    }
```

Comment:
This is the most important change. Feedback welcome. 
First of all, I picked the name "remote action" to mean the actions that don't have an implementation locally. 

The name `console-log2` obviously doesn't exist. I made that up. But I needed to call it something. 

So, `console-log` is specifically problematic because `console-log` is an exact name that exists in **two** places at the moment. 

* Perhaps we should call this one `console-log` and make it override the "local action" called `console-log`. 
* Perhaps we rename `console-log2` to `console-log` the day we do `git rm -r recipe-server/client/actions/console-log/` and make `console-log`, from that day onwards, a remote one. 
* What do you think about the mapping? Somewhere, we need to write down the name of the action as it appears in `mozilla-normandy-action-argument-schemas`.

Reply:
I'm curious about why you chose to implement it this way?

Wouldn't it be better to just add a new field to the `Action` model? Maybe something along the lines of:
```python
remote_mapping = models.CharField(max_length=32, null=True)
```

In this case `remote_mapping` would be set to `'consoleLog'`. And if an action does not have a `remote_mapping` we know it is local.

Reply:
Before the Action can be created, we need to know what it's called in mozilla-central. Technically, what it's called in mozilla-normandy-action-argument-schemas. 

One grander possibility is to NOT have `settings.REMOTE_ACTIONS` at all. Instead, we can do something like this:
```python
with open('mozilla-normandy-action-argument-schemas/schemas.json') as f:
  schemas = json.load(f)
  for key, schema in schemas:  # e.g. prefRollout, consoleLog, etc. 
      try:
          action = Action.objects.get(name=key)
          action.implementation = None
          action.schema = json.dumps(schemas[key])
          action.save()
      except Action.DoesNotExist:
          Action.objects.create(
              name=key,
              schema=json.dumps(schemas[key]),
              implementation=None,  # default anyway
          )
```

Then, to add new actions, you `yarn upgrade mozilla-normandy-action-argument-schemas --latest && ./manage.py update_actions`

This way, the actions are not listed in advance. If we add another action to mozilla-central (and not forget to add its JSON schema to mozilla-normandy-action-argument-schemas`) we won't have to mess with `recipe-server/normandy/settings.py`. The *name* in `Action.name` will be the names as they appear in the JS client code. E.g. `consoleLog` or `preferenceRollout`.

Reply:
> One grander possibility is to NOT have settings.REMOTE_ACTIONS at all.

That was sort of what I was getting at. It seems like it would be pretty annoying to have to maintain this list manually. I think it would be better to just use `update_actions` to generate the list of actions as we do now. And the more I think about it the less we would seem to need a mapping.

Reply:
I like the grand idea of not maintaining a mapping for actions in `settings.py`. I'd go so far as to remove the other mapping too, if we could get away with it.

There is a problem here though. The names "remote actions" and "local actions" are going to get confused really quick. In the client, and in most of the things I've been writing, "remote actions" means an action with an implementation on the server, and "local action" means an action with an implementation in the client. Re-using these names as-is would be confusing, but swapping them on the server is also confusing. I think that we need to make a distinction on the server, it should be between actions with and without implementations. I don't think we should use the words "local" or "remote" at all in the server-side code.

The problem of an action existing both in `mozilla-normandy-action-argument-schemas` and in `mozilla/normandy` is going to come up over and over. Every action we migrate (i.e. all of them) will be in the same situation as `console-log` eventually, and the pattern of `console-log2` isn't going to fly. I've touched on this elsewhere, but one idea that comes to mind is to take the schema from the npm package, but keep any implementation we have locally. This also complicates things though, if we change the schema before fully deprecating the server-side implementation.

I think a good way forward would be to use the new schemas, and then whenever we make new schemas, be careful to make them backwards compatible. Perhaps when updating schemas we should print a warning if it causes any pre-existing (enabled?) recipes to have invalid arguments?